astpretty.py

```diff
@@ -96,6 +96,9 @@ def indented() -> Generator[None, None, None]:
             state.indent -= 1
 
         def indentstr() -> str:
+            if isinstance(indent, int):
+                return state.indent * (' ' * indent)
```

Comment:
this should be resolved earlier so the computation doesn't happen every time -- probably `if isinstance(indent, int):   indent = indent * ' '` at the top -- mypy should understand the type narrowing I think

Reply:
nope, looks like it doesn't -- any other way to convince mypy?

Reply:
are you sure?

```python
from typing import Union

def f(x: Union[int, str]):
    if isinstance(x, int):
        x = x * ' '
    reveal_type(x)


OUTPUT = '''\
$ mypy t.py
t.py:6: note: Revealed type is 'builtins.str'
'''
```

Reply:
putting `reveal_type` just below the `isinstance` statement does give that `indent` is of type `builtins.str`, but

```python
def indentstr() -> str:
            reveal_type(indent)
            return state.indent * indent
```

doing so inside the `indentstr` after the isinstance modification, still reveals it to be union:

```console
❯ mypy --strict astpretty.py
astpretty.py:85: note: Revealed type is 'builtins.str'
astpretty.py:104: note: Revealed type is 'Union[builtins.str, builtins.int]'
astpretty.py:105: error: Incompatible return value type (got "Union[str, int]", expected "str")
Found 1 error in 1 file (checked 1 source file)
```

Reply:
seems like maybe a bug in mypy?

Reply:
Yeah, seems like it. So what's the resolution then? I could cast it or add type ignore maybe, or we could keep the code as-is

Reply:
Another option would be to do the isinstance check in both places for the time being -- the inner isinstance check will always be false, but it's just an extra if check everytime instead of doing a bunch of string manipulation everytime.

Reply:
shrugs, I moved the isinstance check so the check happens less often